### ISSUE TITLE
[a11y] /privacy page accessibility improvements

### ISSUE THREAD

**Issue by alexgibson** [mozilla/bedrock#15343]
# Axe-core® Accessibility Results for /en-US/privacy/

Page URL: https://www-dev.allizom.org/en-US/privacy/

## axe-core found 4 violations

| #   | Description | Axe rule ID | WCAG | Impact | Count |
| --- | --- | --- | --- | --- | --- |
| [1](#1) | Elements must only use supported ARIA attributes | aria-allowed-attr | WCAG 2 Level A, WCAG 4.1.2 | critical | 1   |
| [2](#2) | Heading levels should only increase by one | heading-order | Best practice | moderate | 1   |
| [3](#3) | All page content should be contained by landmarks | region | Best practice | moderate | 2   |

### Failed

1. Elements must only use supported ARIA attributes

[Learn more](https://dequeuniversity.com/rules/axe/4.10/aria-allowed-attr?application=playwright)

aria-allowed-attr

WCAG 2 Level A, WCAG 4.1.2

Ensure an element's role supports its ARIA attributes

critical

Issue Tags: cat.aria wcag2a wcag412 EN-301-549 EN-9.4.1.2

| #   | Issue Description | To solve this violation, you need to... |
| --- | --- | --- |
| 1   | **Element location**<br><br>```<br>.mzp-c-sidemenu-main<br>```<br><br>**Element source**<br><br>```<br><section class="mzp-c-sidemenu-main" aria-expanded="false"><br>``` | Fix all of the following:<br><br>*   ARIA attribute is not allowed: aria-expanded="false" |

2. Heading levels should only increase by one

[Learn more](https://dequeuniversity.com/rules/axe/4.10/heading-order?application=playwright)

heading-order

Best practice

Ensure the order of headings is semantically correct

moderate

Issue Tags: cat.semantics best-practice

| #   | Issue Description | To solve this violation, you need to... |
| --- | --- | --- |
| 1   | **Element location**<br><br>```<br>#contact > h4<br>```<br><br>**Element source**<br><br>```<br><h4>Contact Mozilla</h4><br>``` | Fix any of the following:<br><br>*   Heading order invalid |

3. All page content should be contained by landmarks

[Learn more](https://dequeuniversity.com/rules/axe/4.10/region?application=playwright)

region

Best practice

Ensure all page content is contained by landmarks

moderate

Issue Tags: cat.keyboard best-practice

| #   | Issue Description | To solve this violation, you need to... |
| --- | --- | --- |
| 1   | **Element location**<br><br>```<br>header<br>```<br><br>**Element source**<br><br>```<br><header><br>``` | Fix any of the following:<br><br>*   Some page content is not contained by landmarks |
| 2   | **Element location**<br><br>```<br>.privacy-footnote<br>```<br><br>**Element source**<br><br>```<br><footer class="privacy-footnote"><br>``` | Fix any of the following:<br><br>*   Some page content is not contained by landmarks |

**Comment by clarapenn** [comment]
Hi @alexgibson, no rush to reply but I've begun looking at this one and failure no.1, the aria attribute “expanded = false” being set on the side menu, mzp-c-sidemenu-main by the javascript on the page, is correct for the mobile view but in desktop mode the menu is not hidden, so at this point I think the attribute needs to be “expanded = true”. 

However, changing this looks like it could be a deeper thing than I am currently able to do in terms of where the change would be need to be made without affecting other things, etc. If you can give any pointers I can go ahead, though. Thanks so much!

**Comment by alexgibson** [comment]
@clarapenn you're definitely on the right path! It took me a little while to hunt this one down, and it looks like it's being set in JS here: https://github.com/mozilla/bedrock/blob/main/media/js/base/mozilla-article.js#L13

However to make it a little tricky, the JS is actually from one of our design system components here: https://github.com/mozilla/protocol/blob/main/assets/js/protocol/sidemenu.js

I think what we would need to do it change the component JS in Protocol (our design system), so it has both an `init()` and `destroy()` function, which would then be called in the bedrock code when the page is resized. Does that make sense?

If you wanted to try and take this on, I could open a bug in the Protocol repo for you to fix? then we'd need to publish a new version and pull it into bedrock here to make the fix.
